kube-runtime/src/controller/mod.rs

```diff
@@ -758,6 +758,69 @@ where
         self
     }
 
+    /// Trigger the reconciliation process for a managed object `ObjectRef<K>` whenever `trigger` emits a value
+    ///
+    /// For example, this can be used to watch resources once and use the stream to trigger reconciliation and also keep a cache of those objects.
+    /// That way it's possible to use this up to date cache instead of querying Kubernetes to access those resources
+    ///
+    /// Example:
+    ///
+    /// ```no_run
+    /// # async {
+    /// use futures::{StreamExt, TryStreamExt};
+    /// use k8s_openapi::api::core::v1::{ConfigMap, Pod};
+    /// use kube::api::ListParams;
+    /// use kube::runtime::controller::Action;
+    /// use kube::runtime::reflector::ObjectRef;
+    /// use kube::runtime::{reflector, watcher, Controller, WatchStreamExt};
+    /// use kube::ResourceExt;
+    /// use kube::{Api, Client};
```

Comment:
this will be pretty huge in the docs. i'd probably put the imports in a comment, e.g.:
```suggestion
    /// # use kube::api::{Api, ListParams};
    /// # use kube::runtime::controller::Action;
    /// # use kube::runtime::reflector::ObjectRef;
    /// # use kube::runtime::{reflector, watcher, Controller, WatchStreamExt};
    /// # use kube::{Client, ResourceExt};
```